### PR TITLE
[flang][OpenMP] Re-enable tests when building OpenMP as a runtime

### PR DESCRIPTION
**Pull request by mjklemm**
This PR re-enables OpenMP tests when the OpenMP library is configured as a runtime build.  After PR #80874 has disabled them for a while.

The test system now has a command substitution that adds "-J" with the proper folder that contains omp_lib.{h,mod} to the Flang compiler command line in a test.  This is only done when the OpenMP library is a runtime build (via `LLVM_ENABLE_RUNTIMES`), but is skipped if the OpenMP library is built as a regular LLVM project (via `LLVM_ENABLE_PROJECTS`)

### CODE REVIEW

**Comment by mjklemm** [comment]
This PR cannot land before PR #88934 has landed.

**Comment by banach-space** [comment]
Why not update substitutions for `%flang_fc1`/`%bbc` etc? This way you could avoid updating so many tests. And RUN lines would be shorter.

I don't have a strong opinion. Thinking out loud - the proposed implementation that even people who don't work on OpenMP will have to remember to add this extra variable to RUN lines. Admittedly, updating %flang_fc1`/`%bbc` obfuscates things a bit. But that would only affect OpenMP experts?

**Comment by mjklemm** [comment]
I was playing with the option to extend the `extra_args` field of `ToolSubst`.  But then, if the OpenMP runtime is enabled, it would inject `-J` for all tests not only the OpenMP tests.  I kind of felt that this was a bit too much.  That's why I went with the option of having `%openmp_module_flag` and putting the burden on the OpenMP tests only.  I was hoping that the usual test of creating a new test is by "borrowing" from an existing test and thus automatically getting `%openmp_module_flag` into the new test.

Other options that I see (happy to get more)
- Have an extension to Flang to pick up `-J` and `-I` from the environment (similar to `LIBRARY_PATH` or `INCLUDE_PATH` with C compilers)
- Change the build system to copy/link/move omp_lib.mod from where it lives for runtime builds to where it lives when doing a project build

**Comment by mjklemm** [comment]
> Just a suggestion: we can combine `-J ...` and `-fopenmp`, and name it just `%openmp_flag`. This will still require changes in all tests, but going forward we can add more things in `%openmp_flag`, if needed, and all OpenMP tests will use it instead of `-fopenmp`.

I kind of like that!  I'll craft an update tomorrow and post it here.  Thanks for the suggestion!

**Review by tblah — APPROVED** [review]
Looks good to me. Thanks for the update

**Review by banach-space — APPROVED** [review]
I missed that this change only impacts OpenMP tests. This LGTM, thanks!

**Comment by mjklemm** [comment]
Thanks all for the approval.  I will let this PR sit until the dependent PR has been merged.  Not sure how long that will take.  If it will take too long, I will ask @kparzysz to re-open #88914 and land it.

**Comment by kiranchandramohan** [comment]
All the FIR-based tests (eg flang/test/Lower/OpenMP/FIR/teams.f90) are now removed. You might need a rebase.

**Comment by mjklemm** [comment]
@kiranchandramohan @banach-space I think this go in now.  If you agree, please approve, that I can merge.  The conflicts seem easy to resolve, so I'd do this during merge.

**Review by kiranchandramohan — APPROVED** [review]
LGTM.

**Review by tblah — APPROVED** [review]
Thanks!

File: flang/test/lit.cfg.py
Comment (kkwli) [inline flang/test/lit.cfg.py:207]: build → built

**Review by banach-space — APPROVED** [review]

**Pull request merged**
